controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	clowder "github.com/RedHatInsights/clowder/apis/cloud.redhat.com/v1alpha1"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ClowdenvironmentReconciler reconciles a Clowdenvironment object
+type ClowdenvironmentReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+//+kubebuilder:rbac:groups=cloud.redhat.com,resources=clowdenvironments,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=cloud.redhat.com,resources=clowdenvironments/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=cloud.redhat.com,resources=clowdenvironments/finalizers,verbs=update
+
+func (r *ClowdenvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	env := clowder.ClowdEnvironment{}
+	if err := r.Client.Get(ctx, req.NamespacedName, &env); err != nil {
+		r.Log.Error(err, "Error retrieving clowdenv", "env-name", env.Name)
+		return ctrl.Result{}, err
+	}
+
+	r.Log.Info("Reconciling clowdenv", "env-name", env.Name, "conditions", env.Status.Conditions)
+
+	if ready, _ := VerifyClowdEnvReady(env); ready {
+		r.Log.Info("Clowdenv ready", "env-name", env.Name)
+		ns := env.Spec.TargetNamespace
+		r.Log.Info("Checking for frontend environment", "ns-name", ns)
+		if err := CreateFrontendEnv(ctx, r.Client, ns, env); err != nil {
+			if !errors.IsAlreadyExists(err) {
+				r.Log.Error(err, "Error creating frontend env", "ns-name", ns)
+				UpdateAnnotations(ctx, r.Client, map[string]string{"status": "error"}, ns)
+			}
+		} else {
+			r.Log.Info("Creating frontend environment", "ns-name", ns)
+			UpdateAnnotations(ctx, r.Client, map[string]string{"status": "ready"}, ns)
```

Comment:
"status" is too generic. Update this to something specific (and document it somewhere).